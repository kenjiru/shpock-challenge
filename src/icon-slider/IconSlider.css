.icon-slider {
    display: flex;
    flex-direction: row;
}

.icon-slider .icon {
    width: 24px;
    height: 24px;
}

.icon-slider .slider {
    flex-grow: 2;
    margin: 0 8px;
}