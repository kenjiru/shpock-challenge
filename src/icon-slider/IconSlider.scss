.icon-slider {
    display: flex;
    flex-direction: row;

    .icon {
        width: 24px;
        height: 24px;
    }

    .slider {
        flex-grow: 2;
        margin: 0 8px;
    }
}
