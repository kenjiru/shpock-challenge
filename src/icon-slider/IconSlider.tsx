import * as React from "react";
import { Component, ReactElement } from "react";
import { Slider, SliderProps } from "material-ui";

import Img from "../img/Img";

import "./IconSlider.css";

class IconSlider extends Component<IIconSliderProps> {
    private sliderStyle: Object = {
        marginTop: "3px",
        marginBottom: 0
    };

    public render(): ReactElement<HTMLElement> {
        const {iconLeft, iconRight, ...props} = this.props;

        return (
            <div className="icon-slider">
                <Img className="icon icon-left" src={this.props.iconLeft}/>
                <div className="slider">
                    <Slider sliderStyle={this.sliderStyle} {...props}/>
                </div>
                <Img className="icon icon-right" src={this.props.iconRight}/>
            </div>
        );
    }
}

interface IIconSliderProps extends SliderProps {
    iconLeft: string;
    iconRight: string;
}

export default IconSlider;
