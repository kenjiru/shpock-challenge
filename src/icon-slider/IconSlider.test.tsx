import * as React from "react";
import { shallow } from "enzyme";
import shallowToJson from "enzyme-to-json";

import IconSlider from "./IconSlider";

describe("ErrorMessage", () => {
    it("should render correctly", () => {
        const output = shallow(
            <IconSlider iconRight="right.png" iconLeft="left.png"/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
