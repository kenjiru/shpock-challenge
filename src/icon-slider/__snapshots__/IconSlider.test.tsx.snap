// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render correctly 1`] = `
<div
  className="icon-slider"
>
  <Img
    className="icon icon-left"
    src="left.png"
  />
  <div
    className="slider"
  >
    <Slider
      axis="x"
      disableFocusRipple={false}
      disabled={false}
      max={1}
      min={0}
      required={true}
      sliderStyle={
        Object {
          "marginBottom": 0,
          "marginTop": "3px",
        }
      }
      step={0.01}
      style={Object {}}
    />
  </div>
  <Img
    className="icon icon-right"
    src="right.png"
  />
</div>
`;
